src/plugins/image/display/image-display.js

```diff
@@ -1,48 +1,68 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { ReactCompareSlider, ReactCompareSliderImage } from 'react-compare-slider';
 import classNames from 'classnames';
 import ClientConfig from '../../../bootstrap/client-config';
+import { EFFECT_TYPE, ORIENTATION, SOURCE_TYPE } from '../constants';
 import { inject } from '../../../components/container-context';
 import GithubFlavoredMarkdown from '../../../common/github-flavored-markdown';
 import { sectionDisplayProps, clientConfigProps } from '../../../ui/default-prop-types';
 
-function getSource(type, url, cdnRootUrl) {
-  switch (type) {
-    case 'external':
+function getSource(sourceType, url, cdnRootUrl) {
+  switch (sourceType) {
+    case SOURCE_TYPE.external:
       return url || null;
-    case 'internal':
+    case SOURCE_TYPE.internal:
       return url ? `${cdnRootUrl}/${url}` : null;
     default:
       return null;
   }
 }
 
-function ImageDisplay({ content, clientConfig, githubFlavoredMarkdown }) {
-  const hover = content.hover && (
-    <div className="Image-secondary">
-      <img
-        className={`Image-img u-max-width-${content.maxWidth || 100}`}
-        src={getSource(content.hover.type, content.hover.url, clientConfig.cdnRootUrl)}
-        />
-      <div
-        className="Image-copyrightInfo"
-        dangerouslySetInnerHTML={{ __html: githubFlavoredMarkdown.render(content.hover.text || '') }}
-        />
-    </div>
+const hoverEffect = (content, githubFlavoredMarkdown, clientConfig) => (
+  <div className="Image-secondary">
+    <img
+      className={`Image-img u-max-width-${content.maxWidth || 100}`}
+      src={getSource(content.effect.sourceType, content.effect.sourceUrl, clientConfig.cdnRootUrl)}
+    />
+    <div
+      className="Image-copyrightInfo"
+      dangerouslySetInnerHTML={{ __html: githubFlavoredMarkdown.render(content.effect.text || '') }}
+    />
+  </div>
+);
+
+
+const revealEffect = (content, clientConfig) => {
+  const { effect } = content;
+  return (
+    <ReactCompareSlider
+      position={effect.startPosition}
+      portrait={effect.orientation === ORIENTATION.vertical}
+      itemOne={<ReactCompareSliderImage src={getSource(content.sourceType, content.sourceUrl, clientConfig.cdnRootUrl)} alt="Image one" />}
+      itemTwo={<ReactCompareSliderImage src={getSource(effect.sourceType, effect.sourceUrl, clientConfig.cdnRootUrl)} alt="Image two" />}
+    />
```

Comment:
We also need to show somehow the copyright info for the reveal type of images

Reply:
Shown